ls1intum/Artemis
# `Quiz exercises`: Disable save quiz button if no changes are present

### Checklist
#### General
#### Client

### Motivation and Context

Close #5159. Also a small cleanup of the quiz-exercise-detail-component which had some old and unused code/variables.

### Description

Fixed the issue where the 'saved entity' was not created correctly and therefore the 'Save' button would always be enabled.
Also fixed the issue where even after undoing changes the 'Save' button would still be enabled. Remove the `initCompleted` and the 'entity' variables which aren't used anymore.

There are some cases where making a change and then undoing it will still cause the 'Save' button to be enabled. These are:
- adding/removing categories
- changing the Batch mode type
- changing the content/text of any of the questions

(This can be handled in a future PR if the effort is considered worth it.)

### Steps for Testing

You can just follow the steps in #5159. Additionally, you can test the other fixes:

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the exercises page for a course
4. Click 'Edit' for quiz exercise (if there's no quiz then create one first)
5. Check that the 'Save' button is disabled at first
6. Check that changing some fields enables the 'Save' button
7. Check that after undoing the changes done in the previous step the 'Save' button is disabled again (There are some exceptions described above)

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-detail-component.ts | 80% | 95% |

### Screenshots
